components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/validators/APIResourceScopeValidator.java

```diff
@@ -0,0 +1,243 @@
+
+public class APIResourceScopeValidator extends OAuth2ScopeValidator{
+    private static final Log log = LogFactory.getLog(APIResourceScopeValidator.class);
+    public static final String CHECK_ROLES_FROM_SAML_ASSERTION = "checkRolesFromSamlAssertion";
+    private static final String SCOPE_VALIDATOR_NAME = "Role based api and product scope validator";
+
+    @Override
+    public boolean validateScope(AccessTokenDO accessTokenDO, String resource) throws IdentityOAuth2Exception {
+
+        // Return true if there is no resource to validate the token against.
+        if (resource == null) {
+            return true;
+        }
+
+        //Get the list of scopes associated with the access token
+        String[] scopes = accessTokenDO.getScope();
+
+        //If no scopes are associated with the token
+        if (scopes == null || scopes.length == 0) {
+            return true;
+        }
+
+        String resourceScopes = null;
+        int resourceTenantId = -1;
+
+        boolean cacheHit = false;
+        // Check the cache, if caching is enabled.
+        OAuthCacheKey cacheKey = new OAuthCacheKey(resource);
+        CacheEntry result = OAuthCache.getInstance().getValueFromCache(cacheKey);
+
+        //Cache hit
+        if (result != null && result instanceof ResourceScopeCacheEntry) {
+            resourceScopes = ((ResourceScopeCacheEntry) result).getScope();
+            resourceTenantId = ((ResourceScopeCacheEntry) result).getTenantId();
+            cacheHit = true;
+        }
+
+        // Cache was not hit. So retrieve from database.
+        if (!cacheHit) {
+            Pair<String, Integer> scopeMap = OAuthTokenPersistenceFactory.getInstance().getTokenManagementDAO()
+                    .findTenantAndScopeOfResource(resource);
+
+            if (scopeMap != null) {
+                resourceScopes = scopeMap.getLeft();
+                resourceTenantId = scopeMap.getRight();
+            }
+
+            cacheKey = new OAuthCacheKey(resource);
+            ResourceScopeCacheEntry cacheEntry = new ResourceScopeCacheEntry(resourceScopes);
+            cacheEntry.setTenantId(resourceTenantId);
+            //Store resourceScope in cache even if it is null (to avoid database calls when accessing resources for
+            //which scopes haven't been defined).
+            OAuthCache.getInstance().addToCache(cacheKey, cacheEntry);
+
+        }
+
+        String[] subscriptionDetails =  accessTokenDO.getTokenType().split(":");
+        String subscriptionType = "API";
+        if (subscriptionDetails.length > 1) {
+            subscriptionType = subscriptionDetails[1];
+        } else {
+            log.info("Subscription Type was not set properly in accesstoken data object. So type is considered as "
+                    + "default  'API'.");
+        }
+
+        List<String> tokenScopeList = new ArrayList<>(Arrays.asList(scopes));
+        List<String> resourceScopeList = Arrays.asList(resourceScopes.split(", "));
```

Comment:
lets split the scope using a comma (not with comman and a space)

Reply:
fixed in https://github.com/wso2/carbon-apimgt/pull/6232/commits/555a2ab2fa875057c6fadb316f7e426484108bd3